controllers/windowsmachine/windowsmachine_controller.go

```diff
@@ -192,7 +192,11 @@ func (m *nodeToMachineMapper) Map(object handler.MapObject) []reconcile.Request
 		log.Error(err, "could not get a list of machines")
 	}
 	for _, machine := range machines.Items {
-		if machine.Status.NodeRef != nil && machine.Status.NodeRef.UID == object.Meta.GetUID() {
+		ok := machine.Status.Phase != nil &&
+			len(machine.Status.Addresses) > 0 &&
+			machine.Status.NodeRef != nil &&
+			machine.Status.NodeRef.UID == object.Meta.GetUID()
+		if ok {
```

Comment:
```suggestion
              if !ok {
                  return nil
              }
              return []reconcile.Request{...
```
Less indentation this way.
```

Reply:
That way the function would return at the first machine in the list that is `!ok`, instead of continuing to range through them and only returning nil when none of the machine objects is ok.